unlock-app/src/hooks/useUniswapRoutes.ts

```diff
@@ -24,22 +26,34 @@ export const useUniswapRoutes = ({
   routes,
   enabled = true,
 }: UniswapRoutesOption) => {
+  const { account } = useAuth()
   const web3Service = useWeb3Service()
   return useQuery(
     ['uniswapRoutes'],
     async () => {
       const result = await Promise.all(
         routes.map(async (route) => {
           try {
+            const params = {
+              network: route.network,
+              tokenIn: route.tokenIn,
+              tokenOut: route.tokenOut,
+              amountOut: route.amountOut,
+              recipient: route.recipient,
+            }
             const response = await web3Service.getUniswapRoute({
-              params: {
-                network: route.network,
-                tokenIn: route.tokenIn,
-                tokenOut: route.tokenOut,
-                amountOut: route.amountOut,
-                recipient: route.recipient,
-              },
+              params,
             })
+            const balance = await getAccountTokenBalance(
+              web3Service,
+              account!,
+              route.tokenIn instanceof Token ? route.tokenIn.address : null,
+              route.network
+            )
+            // If the balance is less than the quote, we cannot make the swap.
+            if (parseFloat(balance) < parseFloat(response.quote.toFixed(2))) {
+              throw new Error('Insufficient balance')
```

Comment:
Maybe no need to throw but just console.log?

Reply:
it is inside a try catch so does log and filter the route.